### PR TITLE
PKAs are now as weak in a vacuum as they are in a pressurized environment

### PR DESCRIPTION
**Pull request by TeomanTheGreat**
## About The Pull Request
PKAs now have the same weakness in a vacuum as they have in a pressurized environment 

## Why It's Good For The Game
The current miner antag strat is getting spaceproof and then going to space with PKAs then depressurizing sec to fight them and easily curb stomp them. PKAs are not balanced for non-mining combat. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
PKA works as it should in a pressurized environment

image

PKA has its damage debuff in a vacuum as well

image

PKA works as intended on lavaland
image

</details>

## Changelog
:cl: Phil Smith
balance: PKAs now have the same weakness in a vacuum as they have in a pressurized environment 
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
or just make PKAs balanced on the station z level. I don't think this is the answer

**Comment by TeomanTheGreat** [comment]
> or just make PKAs balanced on the station z level. I don't think this is the answer

If the miner really wants to use his PKA on the station z-level they can steal a fire pin from sec. There is a box of them.

**Comment by PowerfulBacon** [comment]
Problem is they can just emag it (Or should be able to at least), limiting this to just traitor miners which doesn't solve the antga problem you are trying to solve.

**Comment by PowerfulBacon** [comment]
They should be most effective at low pressures and weak at high or no pressure environments.

**Comment by TeomanTheGreat** [comment]
> They should be most effective at low pressures and weak at high or no pressure environments.

Spacing an area will leave a window in which there is low pressure so that wont change much.

**Comment by TeomanTheGreat** [comment]
> Problem is they can just emag it (Or should be able to at least), limiting this to just traitor miners which doesn't solve the antga problem you are trying to solve.

Just tried emagging the PKA. Didn't work.

**Comment by DerpiestBruh** [comment]
People are complaining about PKAs working like they should its unreal lol

**Comment by zeskorion** [comment]
what if we just kinda
rebalance mining stuff numbers to be on-line with station stuff?

**Comment by DerpiestBruh** [comment]
> Also you could just disable the low pressure damage buff on the station z

doesnt make sense

**Comment by DingleFlop** [comment]
Ided, plz nerf

**Comment by Sinestia** [comment]
> They should be most effective at low pressures and weak at high or no pressure environments.

This


Find what pressure lavaland is at and take the absolute value of wherever you are. If it deviates much then make it weak as hell.

**Comment by DerpiestBruh** [comment]
> capable of deleting entire sec force.

lol?

> Mmm yes I love seeing tot skeleton miner in drake armor with PKA, hiero staff, BS crystals, cores, guardian from uplink and whatever other bullshit there is killing seccies in space

cores are a xenobio issue, space adapt is a genetics issue too. Also skill issue.

**Comment by Fronsis** [comment]
> > capable of deleting entire sec force.
> 
> lol?
> 
> > Mmm yes I love seeing tot skeleton miner in drake armor with PKA, hiero staff, BS crystals, cores, guardian from uplink and whatever other bullshit there is killing seccies in space
> 
> cores are a xenobio issue, space adapt is a genetics issue too. Also skill issue.

For cores they meant the legion scores that insta-heals you (and they can buy the stabilized one as well)

They also don't need space adapt since they're a skeleton, i've seen the exact setup he described THREE times, two yesterday and one today, with small variations

Sure, there's a point when certain things are an ''ided'' or a ''skill issue'' moment, but a meth-speed skeleton with ashdrake, hiero staff, kpa, etc it's a little.. unbalanced. It's only fun for le antag that goes around killing all opposing force.

In any case i'm not sure if PKA should be restricted to off station but mostly because they're a very good tool to fight blobs, but... seeing the current antag meta i'd say the bad outweights the good.

**Comment by Sinestia** [comment]
> In any case i'm not sure if PKA should be restricted to off station but mostly because they're a very good tool to fight blobs, but... seeing the current antag meta i'd say the bad outweights the good.

they trivialize blobs in the most braindead way lol

**Comment by Isy232** [comment]
> what if we just kinda
> rebalance mining stuff numbers to be on-line with station stuff?

Holy shit I actually agree with zesko.

Yeah, IF YOUR UP TO IT.

The better solution is to nerf PKA damage but also nerf the health of all lavaland creatures so that the PKA retains its effectiveness on lavaland while being weaker on the station.

Also please take a look at PKA dual wielding. Unless that was removed, miners can dual wield PKA's which is really OP.

**Comment by DerpiestBruh** [comment]
> miners can dual wield PKA's which is really OP.

its really slow to reload and if im not wrong REALLY inaccurate with the gun wielding system

**Comment by PowerfulBacon** [comment]
> > Problem is they can just emag it (Or should be able to at least), limiting this to just traitor miners which doesn't solve the antga problem you are trying to solve.
> 
> Just tried emagging the PKA. Didn't work. 

That's a bug with gun emagging not transferring to firing pins and should be the case. This is subject to change.

**Comment by PowerfulBacon** [comment]
The low pressure buff will be like 3 tiles since outside it is no pressure and inside is high pressure.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
A much nicer solution would be to make it so the damage buff is low pressure but not 0 zero pressure (minimum as well as a maximum).

**Comment by Kratairo** [comment]
People getting mad about security not auto winning every fight, lets just make it so security gets notified who is antag instantly so they can just redtext them right there! Game much more fun that way

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -567,7 +567,7 @@
 	if(!istype(environment))
 		return
 	var/pressure = environment.return_pressure()
-	if(pressure <= LAVALAND_EQUIPMENT_EFFECT_PRESSURE)
+	if(pressure <= LAVALAND_EQUIPMENT_EFFECT_PRESSURE & pressure > 0)
```
Comment (PowerfulBacon) [inline code/__HELPERS/game.dm]: Use a define for this and make it slightly below the minimum lavaland pressure rather than 0 (1kPa shouldnt count too)

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Comment by Fronsis** [comment]
> People getting mad about security not auto winning every fight, lets just make it so security gets notified who is antag instantly so they can just redtext them right there! Game much more fun that way

You certainly fail to see issue of balance that a miner is with the * example gear provided on the comments above with antagonist status* has.

**Comment by AnCopper** [comment]
Yog did this:
https://github.com/yogstation13/Yogstation/pull/15470

**Comment by benbot16** [comment]
Wouldn't this also mess with miners mining on asteroids? It's not too too common, but it is a thing that happens, and there are asteroid monsters that are pretty beefy.

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -567,7 +567,7 @@
 	if(!istype(environment))
 		return
 	var/pressure = environment.return_pressure()
-	if(pressure <= LAVALAND_EQUIPMENT_EFFECT_PRESSURE)
+	if(pressure <= MAXIMUM_LAVALAND_EQUIPMENT_EFFECT_PRESSURE & pressure >= MINIMUM_LAVALAND_EQUIPMENT_EFFECT_PRESSURE)
```
Comment (PowerfulBacon) [inline code/__HELPERS/game.dm]: Wrong operator. This is a bit & operator, not a logical & operator. Use &&

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/__HELPERS/game.dm (hunk 2 above)
Comment (PowerfulBacon) [inline code/__HELPERS/game.dm]:
```suggestion
	if(pressure <= MAXIMUM_LAVALAND_EQUIPMENT_EFFECT_PRESSURE && pressure >= MINIMUM_LAVALAND_EQUIPMENT_EFFECT_PRESSURE)
```

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**